### PR TITLE
STEEL-79 Volume DataSource Documentation

### PR DESCRIPTION
**Pull request by crunis**

### CODE REVIEW

File: website/docs/d/cm_volume.markdown [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+The `filter` block supports:
+
+* `name` - The name of the volume
+
+* `description` - The description of the volume
+
+* `resource_uid` - The resource_uid of the volume
+
+* `deployment_href` - The href of the deployment that contains de volume
```
Comment (mastamark) [inline website/docs/d/cm_volume.markdown]: typo

File: website/docs/d/cm_volume.markdown [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    name = "my_volume"
+  }
+}
+
+output "volume name" {
+  value = "${data.rightscale_volume.crunis_volume.name}"
+}
+
+output "volume ID" {
+  value = "${data.rightscale_volume.crunis_volume.resource_uid}"
```
Comment (mastamark) [inline website/docs/d/cm_volume.markdown]: Can you also use just 'id' here?  (I believe so)

File: website/docs/d/cm_volume.markdown [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+}
+```
+
+## Argument Reference
```
Comment (mastamark) [inline website/docs/d/cm_volume.markdown]: Your example includes 'cloud_href' but is not listed in the argument reference section.

Additionally, can you indicate which of these arguments are required vs. optional in parens so it matches the other datasource docs?

File: website/docs/d/cm_volume.markdown [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+* `name` - The name of the volume
+
+* `description` - The description of the volume
+
+* `resource_uid` - The resource_uid of the volume
+
+* `deployment_href` - The href of the deployment that contains de volume
+
+* `datacenter_href` - The href of the datacenter that holds the volume
```
Comment (mastamark) [inline website/docs/d/cm_volume.markdown]: Example?  Links to docs?

File: website/docs/d/cm_volume.markdown (hunk 2 above)
Comment (crunis) [reply website/docs/d/cm_volume.markdown]: `resource_uid` returns, for example `vol-045e33fd18a746c46`
Just tried `id`, and it returns what is called, I think, the resource href `/api/clouds/1/volumes/63NEHKF8B7RP4`, pinging Adam about this, I thought looking at https://github.com/rightscale/terraform-provider-rightscale/blob/master/rightscale/data_source_volume.go#L75-L112 was the way to found out ALL fields that are returned

File: website/docs/d/cm_volume.markdown [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+data "rightscale_volume" "a_volume" {
+  cloud_href = "/api/clouds/1"
+
+  filter {
+    name = "my_volume"
+  }
+}
+
+output "volume name" {
+  value = "${data.rightscale_volume.crunis_volume.name}"
```
Comment (rshade) [inline website/docs/d/cm_volume.markdown]: should this  be` data.rightscale_volume.a_volume.name`

File: website/docs/d/cm_volume.markdown [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+output "volume ID" {
+  value = "${data.rightscale_volume.crunis_volume.resource_uid}"
```
Comment (rshade) [inline website/docs/d/cm_volume.markdown]: data.rightscale_volume.a_volume.resource_uid

**Comment by crunis** [comment]
All suggestions considered, please take a new fresh look !

**Pull request merged**